ovc5/scripts/install_sd.sh

```diff
@@ -0,0 +1,273 @@
+#!/bin/bash
+
+set -e  # Exit immediately if a command exits with a non-zero status.
+
+# Mounting Vars
+MOUNT_DIR=/mnt/zynq
+BOOT_DIR=$MOUNT_DIR/boot
+ROOT_DIR=$MOUNT_DIR/root
+
+# System Definition Vars
+TEMP_PASSWORD=temppwd
+HOSTNAME=zynq
+
+FIRMWARE_DIR=$(realpath $PWD/..)/firmware
+# Path to the xsa file exported from Vivado.
+VIVADO_PROJECT=$FIRMWARE_DIR/carrier_board
+XSA_PATH=$VIVADO_PROJECT/design_1_wrapper.xsa
+PETALINUX_DIR=$FIRMWARE_DIR/petalinux
+BOOT_FILES_DIR=$PETALINUX_DIR/images/linux
+
+TMP_MOUNT_DIR=/tmp/petalinux_mnt
+
+help () {
+  cat << EOF
+This script is meant to set up and SD card to load Debian Buster and the latest
+OVC 5 software/firmware into the eMMC on the OVC 5 hardware.
+
+THIS WILL INSTALL DEPENDENCIES FROM APT AS IT GOES. MAKE SURE TO REVIEW SCRIPT.
+
+NOTE: Before running this you will need to generate the bitstream in vivado and
+  export the xsa file. To do this follow the steps below:
+
+  1. Open Vivado
+  2. Top Bar: File > Project > Open
+    - $VIVADO_PROJECT/carrier_board.xpr
+  3. Flow Navigator: Program and Debug > Generate Bitstream
+    - Click through the menu without changing defaults. 
+    - This will take a while to complete.
+  4. Top Bar: File > Export > Export Hardware...
+    - 'Output': Include Bitstream
+    - 'Files' should match:
+        $XSA_PATH
+
+usage: ./install_sd.sh <device>
+
+  device: /dev/ device to install to. This is likely \"sdX\" where X is a
+    letter. Check with gparted or dmesg to make sure the wrong device is not
+    selected (this will re-format the drive so it's preferable to get the right
+    device!). Make sure to unmount the drive before running.
+EOF
+}
+
+build_boot_image () {
+  
+  if ! command -v petalinux-boot &> /dev/null
+  then
+    cat << EOF
+Could not find petalinux-boot.
+
+Is petalinux installed? If not, follow these steps:
+  1. Download:
+    https://www.xilinx.com/support/download/index.html/content/xilinx/en/downloadNav/embedded-design-tools.html
+  2. Install (adjust for version changes):
+    ./petalinux-v2020.2-final-installer.run -d petalinux_sdk/ -p "arm aarch64"
+  3. Source:
+    source petalinux_sdk/settings.sh
+
+If installed, make sure petalinux/settings.sh is sourced.
+
+EOF
+    exit
+  fi
+
+  cp $XSA_PATH $PETALINUX_DIR/system.xsa
+  cd $PETALINUX_DIR
+  petalinux-config --get-hw-description --silentconfig
+  petalinux-build
+  petalinux-package \
+    --boot \
+    --force \
+    --fsbl $BOOT_FILES_DIR/zynq_fsbl.elf \
```

Comment:
This fails because the file is called zynqmp_fsbl.elf on my machine.

Reply:
Fixed. I think I copied that file in from somewhere else when I was fiddling with petalinux/vitis.